retrofit/src/main/java/retrofit2/HttpServiceMethod.java

```diff
@@ -53,11 +54,14 @@ static <ResponseT, ReturnT> HttpServiceMethod<ResponseT, ReturnT> parseAnnotatio
         responseType = Utils.getParameterUpperBound(0, (ParameterizedType) responseType);
         continuationWantsResponse = true;
       } else {
-        if (getRawType(responseType) == Call.class) {
+        Class<?> rawType = getRawType(responseType);
+        if (rawType == Call.class || Future.class.isAssignableFrom(rawType)) {
```

Comment:
If we should check RxJava types as well?

Reply:
We can't because the dependency isn't here. And we can't go by string name because of obfuscation possibility. I'm actually okay just doing Call and not doing Future or anything else. I don't think people mix and match other async types with suspend to the degree that they do Call (if at all).